ts/Series/ArcDiagram/ArcDiagramSeries.ts

```diff
@@ -0,0 +1,603 @@
+
+                node.dlBox = {
+                    x: x + width / 2,
+                    y: y,
+                    height: 0,
+                    width: 0
+                };
+
+            } else {
+                node.shapeArgs = {
+                    x,
+                    y,
+                    width,
+                    height,
+                    display: node.hasShape() ? '' : 'none'
+                };
+            }
+
+        } else {
+            node.dlOptions = {
+                enabled: false
+            };
+        }
+    }
+
+    /* eslint-enable valid-jsdoc */
+}
+
+/* *
+ *
+ *  Prototype Properties
+ *
+ * */
+
+interface ArcDiagramSeries {
+    orderNodes: boolean;
+    pointClass: typeof ArcDiagramPoint;
+}
+extend(ArcDiagramSeries.prototype, {
+    orderNodes: false
+});
+
+/* *
+ *
+ *  Registry
+ *
+ * */
+
+declare module '../../Core/Series/SeriesType' {
+    interface SeriesTypeRegistry {
+        arcdiagram: typeof ArcDiagramSeries;
+    }
+}
+ArcDiagramSeries.prototype.pointClass = ArcDiagramPoint;
+SeriesRegistry.registerSeriesType('arcdiagram', ArcDiagramSeries);
+
+/* *
+ *
+ *  Default Export
+ *
+ * */
+
+export default ArcDiagramSeries;
+
+/* *
+ *
+ *  API Options
+ *
+ * */
+
+/**
+ * An `arcdiagram` series. If the [type](#series.arcdiagram.type)
+ * option is not specified, it is inherited from [chart.type](#chart.type).
+ *
+ * @extends   series,plotOptions.arcdiagram
+ * @exclude   dataSorting
+ * @product   highcharts
+ * @requires  modules/sankey
+ * @requires  modules/arc-diagram
+ * @apioption series.arcdiagram
+ */
+
+/**
+ * A collection of options for the individual nodes. The nodes in an arc
+ * diagram are auto-generated instances of `Highcharts.Point`, but options can
+ * be applied here and linked by the `id`.
+ *
+ * @extends   series.arcdiagram.nodes
+ * @type      {Array<*>}
+ * @product   highcharts
+ * @excluding offset
+ * @apioption series.arcdiagram.nodes
+ */
+
+/**
+ * An array of data points for the series. For the `arcdiagram` series
+ * type, points can be given in the following way:
+ *
+ * An array of objects with named values. The following snippet shows only a
+ * few settings, see the complete options set below. If the total number of data
+ * points exceeds the series' [turboThreshold](#series.area.turboThreshold),
+ * this option is not available.
+ *
+ *  ```js
+ *     data: [{
+ *         from: 'Category1',
+ *         to: 'Category2',
+ *         weight: 2
+ *     }, {
+ *         from: 'Category1',
+ *         to: 'Category3',
+ *         weight: 5
+ *     }]
+ *  ```
+ *
+ * @type      {Array<*>}
+ * @extends   series.arcdiagram.data
+ * @product   highcharts
+ * @excluding outgoing, dataLabels
+ * @apioption series.arcdiagram.data
+ */
+
+/**
+ * Individual data label for each node. The options are the same as
+ * the ones for [series.arcdiagram.dataLabels](#series.arcdiagram.dataLabels).
+ *
+ * @apioption series.arcdiagram.nodes.dataLabels
```

Comment:
Is this the only option in `nodes` array? I think something is missing here: http://localhost:9005/highcharts/series.arcdiagram.nodes.dataLabels (e.g. `id`)
![Zrzut ekranu 2022-01-12 o 11 22 28](https://user-images.githubusercontent.com/1453926/149122320-26665d81-3f80-4caf-9017-0baf8500fa70.png)